app/scripts/metamask-controller.js

```diff
@@ -1018,15 +993,25 @@ export default class MetamaskController extends EventEmitter {
       removeIdentity: this.preferencesController.removeAddress.bind(
         this.preferencesController,
       ),
-      setAccountLabel: this.preferencesController.setAccountLabel.bind(
-        this.preferencesController,
-      ),
-      setSelectedAddress: this.preferencesController.setSelectedAddress.bind(
-        this.preferencesController,
-      ),
-      syncIdentities: this.preferencesController.syncAddresses.bind(
-        this.preferencesController,
-      ),
+      setAccountLabel: (address, label) => {
+        const accountToBeNamed =
+          this.accountsController.getAccountByAddress(address);
+        this.accountsController.setAccountName(accountToBeNamed.id, label);
+
+        this.preferencesController.setAccountLabel(address, label);
+      },
+      setSelectedAddress: (address) => {
+        const accountToBeSet =
+          this.accountsController.getAccountByAddress(address);
+        if (accountToBeSet) {
+          this.accountsController.setSelectedAccount(accountToBeSet.id);
+        }
+
+        this.preferencesController.setSelectedAddress(address);
+      },
+      syncIdentities: async (identities) => {
+        this.preferencesController.syncAddresses(identities);
+      },
```

Comment:
Does this need to be `async`? If so should we `await` the call to `syncAddresses`?

``` suggestion
      syncIdentities: async (identities) => {
        await this.preferencesController.syncAddresses(identities);
      },

Reply:
Maybe we have some inconsistency here. I see that `syncAddresses` JSDoc indicates that it returns a promise, but all the operations in the method are synchronous. I'll open another PR to correct the JSDoc

Reply:
https://github.com/MetaMask/metamask-extension/pull/21890

Reply:
Gotcha, so can we take off the `async`?

``` suggestion
      syncIdentities: (identities) => {
        this.preferencesController.syncAddresses(identities);
      },

Reply:
Right, fixed in [5a0b319](https://github.com/MetaMask/metamask-extension/pull/21437/commits/5a0b319ede09e66ca44cec9ff7e622e1878fc94f)